packages/blocks/src/lib/navigation-map/components/obstacle.svelte

```diff
@@ -0,0 +1,63 @@
+
+<script lang='ts'>
+
+import { theme } from '@viamrobotics/prime-core/theme';
+import * as THREE from 'three';
+import { T } from '@threlte/core';
+import type { Obstacle } from '$lib';
+import { view, hovered } from '../stores';
+
+export let obstacle: Obstacle;
+
+let material: THREE.MeshPhongMaterial;
+
+$: name = obstacle.name;
+$: material?.color.set($hovered === obstacle.name
+  ? theme.extend.colors['solar-power']
+  : theme.extend.colors['power-wire']
+);
+
+</script>
+
+{#each obstacle.geometries as geometry, index (index)}
+  <T.Mesh
+    name={name}
+    obstacle={name}
+    lnglat={obstacle.location}
+    rotation.y={geometry.pose.orientationVector.th * THREE.MathUtils.DEG2RAD}
+  >
+    {#if geometry.type === 'box'}
+      {#if $view === '3D'}
+        <T.BoxGeometry
+          computeBounding={name}
+          args={[geometry.length, geometry.width, geometry.height]}
+        />
+      {:else}
+        <T.PlaneGeometry
+          computeBounding={name}
+          args={[geometry.length, geometry.width]}
+        />
+      {/if}
+    {:else if geometry.type === 'sphere'}
+      {#if $view === '3D'}
+        <T.SphereGeometry
+          computeBounding={name}
+          args={[geometry.radius]}
+        />
+      {:else}
+        <T.CircleGeometry
+          computeBounding={name}
+          args={[geometry.radius]}
+        />
+      {/if}
+    {:else if geometry.type === 'capsule'}
+      <T.CapsuleGeometry
+        computeBounding={name}
+        args={[geometry.radius, geometry.length, 16, 32]}
+      />
+    {/if}
+    <T.MeshPhongMaterial
+      bind:ref={material}
+    />
+  </T.Mesh>
+{/each}
```

Comment:
This makes me really want a generic renderer API for geometries like these. We shouldn't have to add a file like this every time we want to add a new visualization type. Fine for now but let's continue the conversation.

Reply:
For sure, curious about your thoughts!